### PR TITLE
dts: riscv: define riscv,machine-timer

### PR DESCRIPTION
**Pull request by gmarull**
All RISC-V platforms with machine timer, except GigaDevice, define its
base address in soc.h. Nowadays we have Devicetree to describe hardware,
so it needs to be used when appropriate. This patch defines the machine
timer for all platforms that define RISCV_MTIME_BASE and
RISCV_MTIMECMP_BASE in soc.h, re-using the same base addresses.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
Can you update the binding to describe the 2 register regions in the 'reg' property description. (And make `reg` required).

Wonder about adding `reg-names` as well??

**Comment by gmarull** [comment]
> Can you update the binding to describe the 2 register regions in the 'reg' property description. (And make `reg` required).
> 
> Wonder about adding `reg-names` as well??

done, using names makes things much better indeed.

**Review by carlocaione — DISMISSED** [review]

**Review by fkokosinski — DISMISSED** [review]

**Review by carlocaione — DISMISSED** [review]

File: dts/bindings/timer/riscv,machine-timer.yaml [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
 
 compatible: "riscv,machine-timer"
 
 include: base.yaml
 
 properties:
+    reg:
+      required: true
+
+    reg-names:
```
Comment (galak) [inline dts/bindings/timer/riscv,machine-timer.yaml:30]: Add in `description` what the `reg-names` values are.

Comment (gmarull) [reply dts/bindings/timer/riscv,machine-timer.yaml:30]: There's an example in the description. Is it a good practice to override descriptions from fields that already do? This is a generic property already documented in `base.yaml`.

File: dts/riscv/telink/telink_b91.dtsi [hunk 2]
```diff
@@ -40,6 +40,15 @@
 			compatible = "mmio-sram";
 		};
 
+		mtimer: timer@e6000000 {
+			compatible = "riscv,machine-timer";
+			reg = <0xe6000000 0x8
+			       0xe6000008 0x8>;
+			interrupts = <7 0>;
```
Comment (carlocaione) [inline dts/riscv/telink/telink_b91.dtsi:47]: what the interrupt `0` is for? That should be the `User software interrupt` but is that used anywhere?

Comment (gmarull) [reply dts/riscv/telink/telink_b91.dtsi:48]: `0` is the priority, unless this platform does it differently

Comment (carlocaione) [reply dts/riscv/telink/telink_b91.dtsi:48]: hahaha, of course. I'll go hiding in shame, see ya.

**Pull request closed without merging**